pygbm/gradient_boosting.py

```diff
@@ -148,9 +148,7 @@ def fit(self, X, y):
         # Subsample the training set for score-based monitoring.
         if do_early_stopping:
             subsample_size = 10000
-            indices = np.arange(X_binned_train.shape[0])
-            if X_binned_train.shape[0] > subsample_size:
-                indices = rng.choice(indices, subsample_size)
+            indices = rng.choice(X_binned_train.shape[0], min(subsample_size, X_binned_train.shape[0]))
             X_binned_small_train = X_binned_train[indices]
             y_small_train = y_train[indices]
```

Comment:
May I suggest the following:

```diff
         if do_early_stopping:
+            n_samples_train = X_binned_train.shape[0]
             subsample_size = 10000
-            indices = rng.choice(X_binned_train.shape[0], min(subsample_size, X_binned_train.shape[0]))
-            X_binned_small_train = X_binned_train[indices]
-            y_small_train = y_train[indices]
+            if n_samples_train > subsample_size:
+                indices = rng.choice(n_samples_train, subsample_size)
+                X_binned_small_train = X_binned_train[indices]
+                y_small_train = y_train[indices]
+            else:
+                X_binned_small_train = X_binned_train
+                y_small_train = y_train
```

so as to actually disable random subsampling on small datasets.

Reply:
yes, makes more sense.